dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/RateLimiterTest.java

```diff
@@ -0,0 +1,37 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import org.testng.annotations.Test;
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+import java.time.Duration;
+import java.lang.InterruptedException;
+
+public class RateLimiterTest extends Thread{
+    
+    @Test
+    public void testConsume() throws InterruptedException{
+        Thread t1 = new Thread(){
```

Comment:
I'd do this with a Callable and a Future: https://www.baeldung.com/java-runnable-callable#2-with-callable

Reply:
You can have callable return a POJO that has an optional exception and returns a state depending on what happen.